File: backend/internal/model/base_data.go
package model

import "time"

// TestItem represents a test/detection item (检测项目).
type TestItem struct {
	ID         uint      `gorm:"primarykey" json:"id"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`

	Name       string  `gorm:"uniqueIndex:idx_item_name;size:200;not null" json:"name"`
	Code       string  `gorm:"uniqueIndex;size:64;not null" json:"code"`
	Category   string  `gorm:"size:100" json:"category"`
	Unit       string  `gorm:"size:50" json:"unit"`
	Method     string  `gorm:"size:200" json:"method"`
	StandardID *uint   `gorm:"index" json:"standard_id"`
	Price      float64 `gorm:"type:decimal(10,2);default:0" json:"price"`
	Status     int     `gorm:"default:1" json:"status"` // 1=active, 0=disabled

	// Associations
	Standard *TestStandard `gorm:"foreignKey:StandardID" json:"standard,omitempty"`
}

// TestStandard represents a test standard/method (检测标准).
type TestStandard struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`

	Name        string    `gorm:"uniqueIndex:idx_std_name;size:200;not null" json:"name"`
	Code        string    `gorm:"uniqueIndex;size:64;not null" json:"code"`
	Issuer      string    `gorm:"size:200" json:"issuer"`       // 发布机构
	Version     string    `gorm:"size:50" json:"version"`       // 版本号
	PublishDate time.Time `json:"publish_date"`
	FilePath    string    `gorm:"size:500" json:"file_path"`    // 标准文件路径（MinIO）
	Status      int       `gorm:"default:1" json:"status"`
}

// Equipment represents a piece of lab equipment (仪器设备).
type Equipment struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`

	Name             string    `gorm:"uniqueIndex:idx_equip_name;size:200;not null" json:"name"`
	Code             string    `gorm:"uniqueIndex;size:64;not null" json:"code"`
	Model            string    `gorm:"size:100" json:"model"`
	Factory          string    `gorm:"size:200" json:"factory"`          // 生产厂家
	CalibrationDate  time.Time `json:"calibration_date"`                 // 校准日期
	NextCalDate      time.Time `json:"next_cal_date"`                    // 下次校准日期
	Status           int       `gorm:"default:1" json:"status"`          // 1=normal, 0=maintenance, 2=retired
}

// Reagent represents a reagent/consumable (试剂耗材).
type Reagent struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`

	Name         string    `gorm:"uniqueIndex:idx_reagent_name;size:200;not null" json:"name"`
	Code         string    `gorm:"uniqueIndex;size:64;not null" json:"code"`
	Spec         string    `gorm:"size:100" json:"spec"`           // 规格
	Manufacturer string    `gorm:"size:200" json:"manufacturer"`   // 生产厂商
	BatchNo      string    `gorm:"size:100" json:"batch_no"`       // 批号
	StockQty     float64   `gorm:"type:decimal(12,2);default:0" json:"stock_qty"`
	Unit         string    `gorm:"size:20" json:"unit"`            // 单位
	ExpireDate   time.Time `json:"expire_date"`
	Status       int       `gorm:"default:1" json:"status"`
}

func (TestItem) TableName() string     { return "test_items" }
func (TestStandard) TableName() string { return "test_standards" }
func (Equipment) TableName() string    { return "equipment" }
func (Reagent) TableName() string      { return "reagents" }

File: frontend/src/api/baseData.ts
import request from './request'

/* ---- Test Items (检测项目) ---- */
export interface TestItem {
  id: number
  name: string
  code: string
  category: string
  unit: string
  method: string
  standard_id: number | null
  price: number
  status: number
  created_at: string
  updated_at: string
  standard?: TestStandard
}

export function getTestItems(params?: {
  keyword?: string
  category?: string
  page?: number
  page_size?: number
}) {
  return request.get('/base-data/items', { params })
}

export function getTestItem(id: number) {
  return request.get(`/base-data/items/${id}`)
}

export function createTestItem(data: {
  name: string
  code: string
  category?: string
  unit?: string
  method?: string
  standard_id?: number | null
  price?: number
}) {
  return request.post('/base-data/items', data)
}

export function updateTestItem(id: number, data: Partial<TestItem>) {
  return request.put(`/base-data/items/${id}`, data)
}

export function deleteTestItem(id: number) {
  return request.delete(`/base-data/items/${id}`)
}

/* ---- Test Standards (检测标准) ---- */
export interface TestStandard {
  id: number
  name: string
  code: string
  issuer: string
  version: string
  publish_date: string
  file_path: string
  status: number
  created_at: string
  updated_at: string
}

export function getTestStandards(params?: {
  keyword?: string
  page?: number
  page_size?: number
}) {
  return request.get('/base-data/standards', { params })
}

export function getTestStandard(id: number) {
  return request.get(`/base-data/standards/${id}`)
}

export function createTestStandard(data: {
  name: string
  code: string
  issuer?: string
  version?: string
  publish_date?: string
  file_path?: string
}) {
  return request.post('/base-data/standards', data)
}

export function updateTestStandard(id: number, data: Partial<TestStandard>) {
  return request.put(`/base-data/standards/${id}`, data)
}

export function deleteTestStandard(id: number) {
  return request.delete(`/base-data/standards/${id}`)
}

/* ---- Equipment (仪器设备) ---- */
export interface Equipment {
  id: number
  name: string
  code: string
  model: string
  factory: string
  calibration_date: string
  next_cal_date: string
  status: number
  created_at: string
  updated_at: string
}

export function getEquipmentList(params?: {
  keyword?: string
  page?: number
  page_size?: number
}) {
  return request.get('/base-data/equipment', { params })
}

export function getEquipment(id: number) {
  return request.get(`/base-data/equipment/${id}`)
}

export function createEquipment(data: {
  name: string
  code: string
  model?: string
  factory?: string
  calibration_date?: string
  next_cal_date?: string
}) {
  return request.post('/base-data/equipment', data)
}

export function updateEquipment(id: number, data: Partial<Equipment>) {
  return request.put(`/base-data/equipment/${id}`, data)
}

export function deleteEquipment(id: number) {
  return request.delete(`/base-data/equipment/${id}`)
}

/* ---- Reagents (物资管理) ---- */
export interface Reagent {
  id: number
  name: string
  code: string
  spec: string
  manufacturer: string
  batch_no: string
  stock_qty: number
  unit: string
  expire_date: string
  status: number
  created_at: string
  updated_at: string
}

export function getReagents(params?: {
  keyword?: string
  page?: number
  page_size?: number
}) {
  return request.get('/base-data/reagents', { params })
}

export function getReagent(id: number) {
  return request.get(`/base-data/reagents/${id}`)
}

export function createReagent(data: {
  name: string
  code: string
  spec?: string
  manufacturer?: string
  batch_no?: string
  stock_qty?: number
  unit?: string
  expire_date?: string
}) {
  return request.post('/base-data/reagents', data)
}

export function updateReagent(id: number, data: Partial<Reagent>) {
  return request.put(`/base-data/reagents/${id}`, data)
}

export function deleteReagent(id: number) {
  return request.delete(`/base-data/reagents/${id}`)
}